src/agentlab/agents/step_agent/browsergym_step_agent.py

```diff
@@ -0,0 +1,146 @@
+            model=self.model,
+            prompt_mode=self.prompt_mode,
+            previous_actions=self.previous_actions,
+            use_dom=self.use_dom,
+            benchmark=self.benchmark,
+            website_name=self.website_name
+        )
+
+
+class BrowserGymStepAgent(Agent):
+    WEBARENA_AGENTS = {
+        "gitlab": "github_agent",
+        "reddit": "reddit_agent",
+        "shopping": "shopping_agent",
+        "shopping_admin": "shopping_admin_agent",
+        "maps": "maps_agent",
+    }
+
+    def __init__(self,
+                 model: ChatModelArgs,
+                 max_actions: int = 10, verbose: int = 0, logging: bool = False,
+                 root_action: str = None,
+                 action_to_prompt_dict: Dict = None,
+                 low_level_action_list: List = None,
+                 prompt_mode: str = "chat",
+                 previous_actions: List = None,
+                 use_dom: bool = False,
+                 benchmark: str = "miniwob",
+                 website_name: str = None
+                 ):
+        match benchmark:
+            case "miniwob":
+                from .prompts.miniwob import step_fewshot_template
+                root_action = "miniwob_agent"
+            case "webarena":
+                from .prompts.webarena import step_fewshot_template
+                root_action = self.WEBARENA_AGENTS[website_name] if website_name in self.WEBARENA_AGENTS else None
+
+        action_to_prompt_dict = {
+            k: v for k, v in step_fewshot_template.__dict__.items() if isinstance(v, dict)}
+
+        self.model = model.make_chat_model()
+        self.use_dom = use_dom
+        self.agent = StepAgent(
+            model=self.model,
+            max_actions=max_actions, verbose=verbose, logging=logging,
+            root_action=root_action,
+            action_to_prompt_dict=action_to_prompt_dict,
+            low_level_action_list=low_level_action_list,
+            prompt_mode=prompt_mode,
+            previous_actions=previous_actions
+        )
+        super().__init__()
+
+    def get_action(self, obs: dict) -> tuple[str, dict]:
+        url = obs["url"] if "url" in obs else None
+        objective = obs["goal"] if "goal" in obs else None
+        if self.use_dom:
+            observation = obs["pruned_html"] if "pruned_html" in obs else None
+        else:
+            observation = obs["axtree_txt"] if "axtree_txt" in obs else None
+        action, _ = self.agent.predict_action(
+            objective=objective, observation=observation, url=url)
+        return self.parse_action(action), {}
+
+    def parse_action(self, action: str) -> str:
+        """Parse the action to a string from BrowserGym action space."""
+        if "click" in action:
+            click_match = re.search(r'click\s*\[(\d+)\]', action, re.DOTALL)
+            bid = click_match.group(1) if click_match else None
+            return f"click(\"{bid}\")"
+
+        if "type" in action:
```

Comment:
Something to note for miniwob: the type action has only two arguments `type [id] [content]`. This is different from webarena, which expects` type [id] [content] [press_enter]`.  

Here's the link to where we parse the miniwob actions: 
https://github.com/asappresearch/webagents-step/blob/a24086f948daf14dd1a034e7f8e7e77953aad41f/src/webagents_step/environment/miniwob.py#L70